storage/namespace.go

```diff
@@ -568,9 +568,7 @@ func (n *dbNamespace) Flush(
 func (n *dbNamespace) NeedsFlush(startInclusive time.Time, endInclusive time.Time) bool {
 	var (
 		blockSize   = n.nopts.RetentionOptions().BlockSize()
-		startBlock  = startInclusive.Truncate(blockSize)
-		endBlock    = endInclusive.Truncate(blockSize)
-		blockStarts = timesInRange(startBlock, endBlock, blockSize)
+		blockStarts = timesInRange(startInclusive, endInclusive, blockSize)
```

Comment:
We assume the input to this method is namespace block aligned. A few changes: 
- We only needed the truncation logic in `namespace.NeedsFlush` because the `databaseCleanupManager` was checking time ranges that were aligned to commit log block times. The new times we generate in the `databaseCleanupManager` are already namespace block aligned, so we don't need the truncation any more. 
- `databaseCleanupManager` was using the `databaseFlushManager`'s `NeedsFlush` method. The `databaseCleanupManager` now checks `databaseNamespace.NeedsFlush`. So I also removed `NeedsFlush` from the `databaseFlushManager`. 
- Changed the `databaseNamespace.NeedsFlush()` method to expect times that are aligned to the namespace block size (the function comment in `types.go` reflects this)